packages/vite/src/vite-node.ts

```diff
@@ -156,13 +159,18 @@ function createViteNodeApp (ctx: ViteBuildContext, invalidates: Set<string> = ne
       if (isAbsolute(moduleId) && !isFileServingAllowed(moduleId, viteServer)) {
         throw createError({ statusCode: 403 /* Restricted */ })
       }
-      const module = await node.fetchModule(moduleId).catch((err) => {
+      const module = await node.fetchModule(moduleId).catch(async (err) => {
         const errorData = {
           code: 'VITE_ERROR',
           id: moduleId,
           stack: '',
+          message: err.message,
           ...err,
         }
+
+        if (!errorData.frame && NEED_FRAME_ERRORS.includes(errorData.code)) {
```

Comment:
I think we can directly compare for a slight perf benefit (and refactor to Set if we need to check more error codes)